src/js/main.js

```diff
@@ -162,6 +162,16 @@ define(function (require, exports) {
 
         /* global _spaces */
         _spaces._debug.enableDebugContextMenu(true, function () {});
+
+        /* global __PG_DEBUG__ */
+        if (__PG_DEBUG__) {
+            // Expose these for snippet usage, only available in debug builds
+            window.__ADAPTER_DESCRIPTOR__ = adapter.ps.descriptor;
+            window.__ADAPTER_OS__ = adapter.os;
+            window.__ADAPTER_UI__ = adapter.ps.ui;
+            window.__FLUX_CONTROLLER__ = _controller;
+            window.__LOG_UTIL__ = log;
```

Comment:
Can we also expose `js/util/performance`? Also, do we need different variables for all the adapter objects? Can we just expose `adapter`?